src/thumbnailer.cc

```
@@ -257,7 +235,38 @@ Thumbnailer::Status Thumbnailer::GenerateAnimationEqualPSNR(
   }
 
   std::cerr << std::endl;
-  std::cerr << "Final PSNR:" << final_psnr << std::endl;
+  std::cerr << "Final PSNR: " << final_psnr << std::endl;
+
+  if (loop_count_ == 0) return kOk;
+  return SetLoopCount(webp_data);
+}
+
+Thumbnailer::Status Thumbnailer::SetLoopCount(WebPData* const webp_data) {
+  WebPMuxError err;
+  WebPMuxAnimParams new_params;
+  uint32_t features;
+
+  std::unique_ptr<WebPMux, void (*)(WebPMux*)> mux(WebPMuxCreate(webp_data, 1),
+                                                   WebPMuxDelete);
+
+  if (mux.get() == nullptr) return kMemoryError;
+
+  if (WebPMuxGetFeatures(mux.get(), &features) != WEBP_MUX_OK) {
+    return kMemoryError;
+  }
+
+  if (WebPMuxGetAnimationParams(mux.get(), &new_params) != WEBP_MUX_OK) {
+    return kMemoryError;
```

It would be better to return the same error that WebPMuxGetAnimationParams gave. You could define a macro, like (I adapted it from one we have in webp2):
```
#define CHECK_MUX_STATUS(status)              \
  do {                                        \
    const WebPMuxError __S__ = (status);         \
    if (__S__ != WEBP_MUX_OK) {             \
      return __S__;                           \
    }                                         \
  } while (0)
```

Then you can just do:
CHECK_MUX_STATUS(WebPMuxGetAnimationParams(mux.get(), &new_params));

---

But `WebPMuxGetAnimationParams()` returns a `WebPMuxError`, do you suggest changing the returned type (currently is `Thumbnailer::Status`)? Maybe the macro needs a `WebPMuxError` to `Thumbnailer::Status` conversion instead

---

Yeah sorry I didn't realize we were using different error types. But I was just thinking that always returning kMemoryError even if we get an unrelated WebPMuxError (e.g. WEBP_MUX_INVALID_ARGUMENT, WEBP_MUX_BAD_DATA) is not great. So yeah a function/macro to convert between WebPMuxError and Thumbnailer::Status would be good, but probably in another PR.